CONTRIBUTING.md

```diff
@@ -495,6 +495,31 @@ In order to help the decision, CI automatically assigns [`rebuild` labels](https
 As a rule of thumb, if the number of rebuilds is **500 or more**, consider targeting the `staging` branch instead of `master`; if the number is **1000 or more**, the pull request causes a mass rebuild, and should target the `staging` branch.
 See [previously merged pull requests to the staging branches](https://github.com/NixOS/nixpkgs/issues?q=base%3Astaging+-base%3Astaging-next+is%3Amerged) to get a sense for what changes are considered mass rebuilds.
 
+Please note that changes to the Linux kernel are an exception to this rule. These PRs go to `batch-with-kernel-updates`,
+see [the next section for more context](#changes-rebuilding-all-tests).
+
+### Changes rebuilding all tests
+[test-driver-rebuild]: #changes-rebuilding-all-tests
+
+Changes causing a rebuild of all tests get a special [`10.rebuild-all-nixos-tests`](https://github.com/NixOS/nixpkgs/issues?q=state%3Aopen%20label%3A10.rebuild-nixos-tests) label.
+
+This means that all VM tests get rebuild which poses a significant impact on the build infrastructure.
+Hence, these PRs should either target a `staging`-branch or `batch-with-kernel-updates`, provided the following conditions apply:
+
+* The label `10.rebuild-all-nixos-tests` is set, or
+* The PR is a change affecting the Linux kernel.
```

Comment:
Don't we only need the former condition?

Reply:
But wouldn't those kernel builder changes also get the label?

Reply:
As mentioned in https://github.com/NixOS/nixpkgs/pull/444973#discussion_r2394529565, I think we have good reasons to do it like this, hence resolving.